packages/playwright-core/src/server/dispatchers/dispatcher.ts

```diff
@@ -224,6 +233,31 @@ export class DispatcherConnection {
     throw new ValidationError(`${path}: expected dispatcher ${names.toString()}`);
   }
 
+  registerDispatcher(dispatcher: DispatcherScope) {
+    this._dispatchers.set(dispatcher._guid, dispatcher);
+
+    let list = this._dispatchersByType.get(dispatcher._type);
+    if (!list) {
+      list = new Set();
+      this._dispatchersByType.set(dispatcher._type, list);
+    }
+    list.add(dispatcher._guid);
+    if (list.size > maxDispatchers)
+      this._disposeStaleDispatchers(list);
+  }
+
+  private _disposeStaleDispatchers(dispatchers: Set<string>) {
+    const dispatchersArray = [...dispatchers];
+    const collect = dispatchersArray.splice(0, maxDispatchers / 10);
+    dispatchers = new Set(dispatchersArray.slice(maxDispatchers / 10));
```

Comment:
What does this line do? Did you mean to mutate `dispatchers` set in-place? Probably not!